src/cmd/metadata.rs

```diff
@@ -63,14 +61,19 @@ impl GenerateMetadataCommand {
         let fname_bundle = format!("{}.contract", self.crate_metadata.package_name);
         let out_path_bundle = cargo_meta.target_directory.join(fname_bundle);
 
-        let target_dir = cargo_meta.target_directory.clone();
+        let target_directory = cargo_meta.target_directory.clone();
 
         // build the extended contract project metadata
-        let (dest_wasm, source_meta, contract_meta, user_meta, optimization_result) =
-            self.extended_metadata()?;
+        let ExtendedMetadataResult {
+            dest_wasm,
+            source,
+            contract,
+            user,
+            optimization_result,
+        } = self.extended_metadata()?;
```

Comment:
My preference would be to use the result direactly to access the fields e.g. `result.dest_wasm`, instead of destructuring it. I can see that it has benefits in the construction of the `GenerationResult` below. That's only my preference though so if you prefer this then :+1:

Reply:
Okay, I'll leave it as is then, since it keeps the subsequent code shorter and has the benefits you mentioned.